### DISCUSSION TITLE
Excessive amount of GUI notifications in --monitor mode [Ubuntu 24.04]

### DISCUSSION THREAD

**Discussion by nicoloesch** [abraunegg/onedrive#2894] [Q&A] [answered]
Hi everyone,

First of all, I want to thank @abraunegg for this amazing implementation of OneDrive for Linux. I have been using it since 3-4 years and it has been working like a charm.
Recently, I updated my personal laptop to Ubuntu 24.04. Since then, I have received an excessive amount of GUI notifications:

![onedrive_notification](https://github.com/user-attachments/assets/0a2308c8-4b9c-4998-83cf-482e8774c133)

I was wondering why these notifications are that frequent, with some syncs starting every minute.
I followed the installation as outline in the Documentation and setup the `onedrive --monitor` as a [Systemd service](https://github.com/abraunegg/onedrive/blob/master/docs/usage.md#onedrive-service-running-as-root-user-via-systemd-arch-ubuntu-debian-opensuse-fedora).

My configuration of the client includes [what I assume to be the most important parameters - the full config is at the end]:
- `monitor_interval`: 300
- `version`: v2.5.2-1+np1+1.1

As my `monitor_interval` does not match the GUI notification interval, I thought I would ask this question in here. Preferably, I would like to get notifications if a file sync was unsuccesful but I do not need to be aware of when onedrive starts its sync. I am aware that this is a very specific user request and it might not be possible as I have only stumbled over the option to entirely disable notifications.

Thank you for your help!
Nico

```
---- FULL CONFIG (`onedrive --display-config` ---------
Reading configuration file: /home/nico/.config/onedrive/config
Configuration file successfully loaded
D-Bus message bus daemon is available; GUI notifications are now enabled
Application version                          = onedrive v2.5.2-1+np1+1.1
Compiled with                                = LDC 2106
User Application Config path                 = /home/nico/.config/onedrive
System Application Config path               = /etc/onedrive
Applicable Application 'config' location     = /home/nico/.config/onedrive/config
Configuration file found in config location  = true - using 'config' file values to override application defaults
Applicable 'sync_list' location              = /home/nico/.config/onedrive/sync_list
Applicable 'items.sqlite3' location          = /home/nico/.config/onedrive/items.sqlite3
Config option 'drive_id'                     = 
Config option 'sync_dir'                     = ~/OneDrive
Config option 'enable_logging'               = false
Config option 'log_dir'                      = /var/log/onedrive
Config option 'disable_notifications'        = false
Config option 'skip_dir'                     = University/PhD/03_Code/
Config option 'skip_dir_strict_match'        = false
Config option 'skip_file'                    = ~*|.~*|*.tmp|*.swp|*.partial
Config option 'skip_dotfiles'                = false
Config option 'skip_symlinks'                = false
Config option 'monitor_interval'             = 300
Config option 'monitor_log_frequency'        = 12
Config option 'monitor_fullscan_frequency'   = 12
Config option 'read_only_auth_scope'         = false
Config option 'dry_run'                      = false
Config option 'upload_only'                  = false
Config option 'download_only'                = false
Config option 'local_first'                  = false
Config option 'check_nosync'                 = false
Config option 'check_nomount'                = false
Config option 'resync'                       = false
Config option 'resync_auth'                  = false
Config option 'cleanup_local_files'          = false
Config option 'classify_as_big_delete'       = 1000
Config option 'disable_upload_validation'    = false
Config option 'disable_download_validation'  = false
Config option 'bypass_data_preservation'     = false
Config option 'no_remote_delete'             = false
Config option 'remove_source_files'          = false
Config option 'sync_dir_permissions'         = 700
Config option 'sync_file_permissions'        = 600
Config option 'space_reservation'            = 52428800
Config option 'application_id'               = d50ca740-c83f-4d1b-b616-12c519384f0c
Config option 'azure_ad_endpoint'            = 
Config option 'azure_tenant_id'              = 
Config option 'user_agent'                   = ISV|abraunegg|OneDrive Client for Linux/v2.5.2-1+np1+1.1
Config option 'force_http_11'                = false
Config option 'debug_https'                  = false
Config option 'rate_limit'                   = 0
Config option 'operation_timeout'            = 3600
Config option 'dns_timeout'                  = 60
Config option 'connect_timeout'              = 10
Config option 'data_timeout'                 = 60
Config option 'ip_protocol_version'          = 0
Config option 'threads'                      = 8
Environment var 'XDG_RUNTIME_DIR'            = true
Environment var 'DBUS_SESSION_BUS_ADDRESS'   = true
Config option 'notify_file_actions'          = false

Selective sync 'sync_list' configured        = false

Config option 'sync_business_shared_items'   = false

Config option 'webhook_enabled'              = false
```

**Comment by abraunegg** [comment, marked as answer]
@nicoloesch 
> I was wondering why these notifications are that frequent, with some syncs starting every minute.

Based on your screen shot  - either you have many systemd services and they are all restarting at the same time ... or systemd is restarting your service every minute .. meaning you have a problem you need to track down.

You will need to most likely enable verbose logging and look at your systemd service logs to work out why.

> As my monitor_interval does not match the GUI notification interval, I thought I would ask this question in here.

The application has a minimum enforced value of 300.

If your GUI configuration has a different value - and is less than 300 it will be ignored.

If this is greater than 300 - then how you are using `--display-config` needs to also include your GUI configuration path so that this command.is reading the right config to display. Please read the documentation on this.

**Reply by nicoloesch** [reply]
Thank you so much for the information! I'll try to track it down!

**Comment by nicoloesch** [comment]
For anyone wondering, I had an issue with an automatically generated folder titled `~` (I do not know what process created this but **definitely not** the `onedrive` client). This resulted in a synchronisation error, that was easily highlighted using `journalctl --user-unit=onedrive -f` (might require adaptation depending on how you set-up the systemd configuration). 
Verifying the folder (and deleting it as it was unnecessary), followed by a `--resync` made all notifications disappear and the client is now operating as intended.

Thanks again @abraunegg  for the help!
